### ISSUE TITLE
WG brand update proposal.

### ISSUE THREAD

**Issue by AlveLarsson** [rust-gamedev/wg#97]
Hello!
I'd like to propose an updated logo/brand

Why?
The current logo has a bit too much going on, 
this aims to simplify and dial back the contrast of the colors. while also setting a clean bold font.

I would recommend this as a set standard, this would help achieve consistency more easily, build trust and authority.
![Group 245 (1)](https://user-images.githubusercontent.com/61861940/101296447-ffb51380-3823-11eb-96e9-381f21b3e5c7.png)

Foreground color: # 191919 
Background color: # E4E4E4
Roboto Black 900: https://fonts.google.com/specimen/Roboto#license (Apache License, Version 2.0.)

Hope you all like it! 🦀 🦀
I would like to take a look at the CSS of the pages next if you like this proposal (I have a lot of time :P).

**Comment by zicklag** [comment]
I like it! I think it looks very professional. It could *maybe* use a little color if it went on the Rust gamedev newsletter site, just to keep things from being 100% monochrome, but I think it's a good direction from a design standpoint.

**Comment by 17cupsofcoffee** [comment]
Kinda agree with Zicklag that it might want a little more color, but the direction is good 👍 

With regards to the site: the design I did is mostly just tweaks to [an existing Jekyll theme](https://github.com/jekyll/minima), so if someone can come up with something better, I'd be on board with that 😄 

The only aspects of the current site I *am* attached to are the performance and accessibility - I put quite a bit of effort in on both fronts and wouldn't want that to be lost in service of a redesign. But beyond that, I'm not picky!

**Comment by AlveLarsson** [comment]
Awesome!
yeah that's the benefit of doing a monochrome design first, it allows for color when you want it, but you have to be careful and ensure it always passes the contrast standards (WCAG AAA) for Accessibility reasons, see https://webaim.org/resources/contrastchecker/

I'm going to see what I can create, do you want to try a rustic color? @17cupsofcoffee @zicklag

> With regards to the site: the design I did is mostly just tweaks to an existing Jekyll theme, so if someone can come up with 
something better, I'd be on board with that 😄

Sign me up, I'm going to take a look at it after this, my first things to change is to add a bit more spacing, as it's quite dense, and unlock the max-width of the page a bit.

**Comment by zicklag** [comment]
Maybe Rustic color would be good. I didn't have any particular ideas. Also it may not *need* the color in the logo or logo banner, but maybe just in the surrounding site. Like maybe make web links orangeish? Not sure that could be a bad idea.

Maybe an idea around the logo is that a touch of red for the buttons or the joystick could be good, but not sure that's a good idea either.

Just some ideas!

**Comment by 17cupsofcoffee** [comment]
Yeah, I suppose the current site doesn't have a *ton* of built-in color - it's mainly just the links/buttons and the big banner on the homepage. On the newsletters, adding colour is probably even less important cause there's a ton of pictures/content to focus on already. So I think the monocrome stuff is definitely a good starting point :+1:

**Comment by AlveLarsson** [comment]
Here is a rust-colored version.
![Group 247 (3)](https://user-images.githubusercontent.com/61861940/101501828-7e1ecc00-3970-11eb-8925-60204c80c99a.png)

> Maybe an idea around the logo is that a touch of red for the buttons or the joystick could be good, but not sure that's a good idea either.

heh, that's usually a trap it could limit where the logo could be placed, having just a monochrome base and shifting the general background and foreground works great.

**Comment by zicklag** [comment]
> heh, that's usually a trap it could limit where the logo could be placed

Yeah, good point.

I think the touch of color did help. It's low enough saturation that you almost don't think about it, but doesn't invoke that black-and-white feeling.

**Comment by AlveLarsson** [comment]
I did try a mutch stronger color but it's kinda hard to get it good looking

**Comment by zicklag** [comment]
Yeah, I think that's good.

**Comment by ozkriff** [comment]
> The current logo has a bit too much going on,

I'm actually not 100% sure if we're talking about replacing the gamepad-in-a-gear logo (https://rust-gamedev.github.io/logo.png) or these icons on the site:

![image](https://user-images.githubusercontent.com/662976/101504975-dfe93100-3984-11eb-8386-73b844103b88.png)

**Comment by ozkriff** [comment]
Also, it might be good to keep in mind https://github.com/rust-gamedev/rust-gamedev.github.io/issues/277 - it would be cool to have a unified style between https://arewegameyet.rs and https://rust-gamedev.github.io

**Comment by AlveLarsson** [comment]
> I'm actually not 100% sure if we're talking about replacing the gamepad-in-a-gear logo

yeah, it's the global brand I'm talking about, the icons on the main page I can do a replacement for later.

> it would be cool to have a unified style between

good point, have been thinking about it a bit, a good solution would maybe be either a merge and do it as a sub-page, then have the URL re-direct to the subpage

another idea is to create a template repo and just having pure site content in the repos like with the .mb files

**Comment by 17cupsofcoffee** [comment]
I like the idea of shared styles/assets more than merging the two sites, at least as a starting point - AreWeGameYet is a lot more complex than the WG site (due to all the calls it makes to external APIs) and I feel like that complexity bleeding into the place where people contribute newsletter posts wouldn't be ideal. Plus I think if every newsletter contributor starts running the AWGY build we'll hit every rate limit under the sun 😆

**Comment by AlveLarsson** [comment]
Oh right... build times and limits is a thing... 😆 
yeah then a root template sounds better

**Comment by AlveLarsson** [comment]
If you want I can start looking into setting up a template with the style set of the proposed brand, I'm personally a massive svelte/ sapper jamstack fan

**Comment by 17cupsofcoffee** [comment]
I personally wouldn't mind switching static site generators, but bear in mind that most of the people who work on/contribute to these sites aren't web developers - ideally we should keep the amount of tooling/web knowledge needed to a minumum. Would be interested to hear the other WG members' opinions on that (cc @ozkriff in particular).

I do definitely think that AWGY is going to have to switch away from Zola at some point though, as the current method of fetching the 3rd party data is really hacky (API calls in templates, ahhhhh 😱) and does way too many individual requests (I've hit rate limits while developing locally before). But that's an issue for that repo, not here!

I've not used Svelte or Sapper myself but I've heard good things, I'll look into those for my own site if nothing else :p

**Comment by ozkriff** [comment]
> yeah, it's the global brand I'm talking about

Oh, in this case I'm not a fan of the proposed logo: getting rid of the iconic rust gear reduces the recognition of belonging to the Rust ecosystem.

**Comment by AlveLarsson** [comment]
> getting rid of the iconic rust gear 

well, it's not gone, it's still the gear as the base, but let's try adding back some of the parts to see what happens. I also cleaned up so there are fewer lines on the joystick.
![Group 256](https://user-images.githubusercontent.com/61861940/101526944-a79a2080-398d-11eb-9967-26432ec04700.png)

**Comment by aclysma** [comment]
> getting rid of the iconic rust gear reduces the recognition of belonging to the Rust ecosystem

Agreed with this - so far I lean towards wanting to keep the iconic sprocket. That said, the most recent edit definitely makes it more obvious that there is a "gear" reference :)

> ideally we should keep the amount of tooling/web knowledge needed to a minumum

I lean towards going whatever makes the people doing the most work with the sites happiest. I think there is value that zola is rust, so it's less intimidating to me knowing rust that I could just go read the zola code if I wanted to figure out how something worked.

> the current method of fetching the 3rd party data is really hacky (API calls in templates, ahhhhh 😱)

At the risk of getting off topic, maybe a separate tool to scrape the data into json, and then generate the templates from that json would make it easier to work with? (Might also make it easier to switch template systems in the future)

> I like the idea of shared styles/assets more than merging the two sites, at least as a starting point

I'm happy with whatever @ozkriff , @17cupsofcoffee, and others who work with those sites want to do - would keeping two smaller projects instead of one bigger one make it easier to spread responsibility/"ownership" among more people? Or does it just get in the way? (My angle on this is wanting to keep things sustainable!)

**Comment by 17cupsofcoffee** [comment]
I also quite like the additions to the new version, keeps the 'feel' of the Rust branding a bit more :smile:

> I lean towards going whatever makes the people doing the most work with the sites happiest. I think there is value that zola is rust, so it's less intimidating to me knowing rust that I could just go read the zola code if I wanted to figure out how something worked.

Yeah, this is my main concern with any tooling changes - the majority of the content on the site comes from PRs by Rust devs (because otherwise Ozkriff would have to write the entire newsletter on his own and he'd probably go crazy 😆 he does enough work as it is!) and I don't want people to feel like they can't contribute without understanding the web ecosystem.

> At the risk of getting off topic, maybe a separate tool to scrape the data into json, and then generate the templates from that json would make it easier to work with? (Might also make it easier to switch template systems in the future)

That's not a bad idea, actually. I'll raise an issue for that on AWGY's tracker.

**Comment by AlveLarsson** [comment]
Idk if rust has the sprocket by itself as an SVG, I did not find anything so I hand made my own rust sprocket so it's in SVG.

![Group 267 (5)](https://user-images.githubusercontent.com/61861940/101564151-e69a9700-39ca-11eb-8d12-82dd8664201e.png)

Here are some ideas, this just may end up being a refinement of the current logo with the added colors and font, I think this is good and just doing away with the excessive curves on the gamepad, and the button complexity for smaller screens.

(btw having this as an SVG makes it great for print on t-shirts, I would actually consider buying one myself if there was a t-shirt shop like what I did for amethyst😉😉)

**Comment by 17cupsofcoffee** [comment]
I really like those new designs, they keep the style of the old one while being a bit more legible at smaller sizes - and having an SVG is always nice :) I think the joystick one and the two controllers with icons on them are my favourites. The controller without the buttons is quite nice and clean too, but I'm not sure if it's quite as readable as 'controller' at first glance.

**Comment by AlveLarsson** [comment]
I would do,

1 ) do a quick replacement for the logo if it works; otherwise, just let it be while doing the new template.

2 ) do the template repo to isolate all the technical website issues to one repo, this would cut the development time and only one source, also don't get me wrong here rust is excellent, but I think it's more appropriate to use more proven technologies for the website than some flashy new rust lib that, I bet a lot more people know regular JS HTML and CSS (oh btw sapper is real simple to use😂 I promise I'm not paid for saying that)
